code/drasil-gool/GOOL/Drasil/Helpers.hs

```diff
@@ -80,8 +80,23 @@ onStateList f as = f <$> sequence as
 on2StateLists :: ([a] -> [b] -> c) -> [State s a] -> [State s b] -> State s c
 on2StateLists f as bs = liftM2 f (sequence as) (sequence bs)
 
-on1StateValue1List :: (a -> [b] -> c) -> State s a -> [State s b] -> State s c
-on1StateValue1List f a as = liftM2 f a (sequence as)
+on1StateWrapped :: (Monad m) => (a -> m b) -> m a -> m b
+on1StateWrapped f a' = do 
```

Comment:
Ah, that's not what I meant!  I meant to use `do` notation in the code itself, **instead** of using `on1StateWrapped`, `on2StateWrapped`.